modules/components/pages/outputs/timeplus.adoc

```diff
@@ -0,0 +1,317 @@
+    apikey: "" # No default (optional)
+    username: "" # No default (optional)
+    password: "" # No default (optional)
+    max_in_flight: 64
+    batching:
+      count: 0
+      byte_size: 0
+      period: ""
+      check: ""
+      processors: [] # No default (optional)
+```
+--
+======
+
+
+This output only accepts structured messages. All messages must:
+
+- Contain the same keys.
+- Use a structure that matches the schema of the destination data stream.
+
+If your upstream data source or pipeline returns unstructured messages, such as strings, you can configure a output processor to transform the messages. See the <<unstructured-message,Unstructured message>> section for examples.
+
+== Examples
+
+[tabs]
+======
+Timeplus Enterprise (Cloud)::
++
+--
+You must https://docs.timeplus.com/apikey[generate an API key^] using the web console of Timeplus Enterprise (Cloud).
+
+```yaml
+output:
+  timeplus:
+    workspace: <workspace-id>
+    stream: <stream-name>
+    apikey: <api-key>
+```
+Replace the following placeholders with your own values:
+
+- `<workspace-id>`: The ID of the workspace you want to send messages to.
+- `<stream-name>`: The name of the destination data stream.
+- `<api-key>`: The API key for the Ingest API.
+
+--
+
+Timeplus Enterprise (Self-Hosted)::
++
+--
+
+You must specify the username, password, and URL of the application server.
+
+```yaml
+output:
+  timeplus:
+    url: http://localhost:8000
+    workspace: <workspace-id>
+    stream: <stream-name>
+    username: <timeplus-username>
+    password: <timeplus-password>
+```
+Replace the following placeholders with your own values:
+
+- `<workspace-id>`: The ID of the workspace you want to send messages to.
+- `<stream-name>`: The name of the destination data stream.
+- `<timeplus-username>`: The username for the Timeplus application server.
+- `<timeplus-password>`: The password for the Timeplus application server.
+
+--
+
+timeplusd::
++
+--
+
+You must specify the HTTP port for the `timeplusd` server process.
+
+```yaml
+output:
+  timeplus:
+    url: http://localhost:3218
+    stream: <stream-name>
+    username: <timeplus-username>
+    password: <timeplus-password>
+```
+Replace the following placeholders with your own values:
+
+- `<stream-name>`: The name of the destination data stream.
+- `<timeplus-username>`: The username for the Timeplus application server.
+- `<timeplus-password>`: The password for the Timeplus application server.
+
+--
+======
+
+
+=== Unstructured messages
+
+If your upstream data source or pipeline returns unstructured messages, such as strings, you can configure an output processor to transform them into a structured messages and then pass them to the output.
+
+This example creates structured messages using the `raw` field, and stores the original string content into it. You can modify the name of the field, but you must also add either the the `raw` field name or your custom field name to the destination data stream.
```

Comment:
Is `raw` supposed to be one of the fields listed on this doc? And am I understanding correctly that `raw` is like a function that parses and transforms values stored in it?

Reply:
No `raw` is just an example field name, and my understanding that yes `raw` is a Blobland function that parses and transforms the string value. I'll try to reword to make that clearer.

Reply:
@kbatuigas, I've tried to reword the text explaining the example. Is it any better now?

Reply:
@kbatuigas yes that looks good to me. Thank you!